src/eos-updater-util.c

```diff
@@ -160,148 +153,165 @@ static const gchar *dmi_attributes[] =
-  gs_free gchar *uri = NULL;
-  gs_free gchar *query = NULL;
-  gs_unref_object OstreeDeployment *merge_deployment = NULL;
-  const gchar *osname;
-  const gchar *booted;
-  GKeyFile *origin;
-  GKeyFile *repo_config;
-  GHashTable *hw_descriptors = NULL;
-  gs_unref_object SoupSession *soup = NULL;
-  gs_unref_object SoupMessage *msg = NULL;
-  gs_unref_keyfile GKeyFile *bkf = NULL;
-
-  OstreeSysroot *sysroot = ostree_sysroot_new_default ();
+  gs_unref_object OstreeSysroot *sysroot = ostree_sysroot_new_default ();
+  OstreeDeployment *booted_deployment = NULL;
 
   if (!ostree_sysroot_load (sysroot, NULL, error))
-    goto out;
+    return NULL;
 
-  if (!ostree_sysroot_get_booted_deployment (sysroot))
+  booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
+  if (booted_deployment == NULL)
     {
       g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
                            "Not an ostree system");
-      goto out;
+      return NULL;
     }
 
-  merge_deployment = ostree_sysroot_get_merge_deployment (sysroot, NULL);
-  osname = ostree_deployment_get_osname (merge_deployment);
-  origin = ostree_deployment_get_origin (merge_deployment);
-  booted = ostree_deployment_get_csum (merge_deployment);
+  return g_object_ref (booted_deployment);
+}
+
+static gchar *
+get_booted_checksum (OstreeDeployment *booted_deployment)
+{
+  return g_strdup (ostree_deployment_get_csum (booted_deployment));
+}
+
+static gboolean
+get_origin_refspec (OstreeDeployment *booted_deployment,
+                    gchar **remote,
+                    gchar **ref,
+                    GError **error)
+{
+  GKeyFile *origin;
+  gs_free gchar *refspec = NULL;
 
-  if (!origin)
+  origin = ostree_deployment_get_origin (booted_deployment);
+
+  if (origin == NULL)
     {
+      const gchar *osname;
+      const gchar *booted;
+
+      osname = ostree_deployment_get_osname (booted_deployment);
+      booted = ostree_deployment_get_csum (booted_deployment);
       g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
                    "No origin found for %s (%s), cannot upgrade",
                    osname, booted);
-      goto out;
+      return FALSE;
     }
 
-  o_refspec = g_key_file_get_string (origin, "origin", "refspec", NULL);
+  refspec = g_key_file_get_string (origin, "origin", "refspec", error);
+  if (refspec == NULL)
+    return FALSE;
+
+  return ostree_parse_refspec (refspec, remote, ref, error);
+}
+
+static gchar *
+get_baseurl (OstreeDeployment *booted_deployment,
+             OstreeRepo *repo,
+             GError **error)
+{
+  const gchar *osname;
+  gs_free gchar *url = NULL;
+
+  osname = ostree_deployment_get_osname (booted_deployment);
+  if (!ostree_repo_remote_get_url (repo, osname, &url, error))
+    return NULL;
+
+  return g_steal_pointer (&url);
+}
 
-  if (!ostree_parse_refspec (o_refspec, &o_remote, &o_ref, error))
-    goto out;
+#define EOS_VENDOR_KEY "sys_vendor"
+#define EOS_PRODUCT_KEY "product_name"
```

Comment:
I'll rename it to `VENDOR_KEY`. The value also originates from device-tree stuff, so it is not really DMI-specific.